File: Diary10/app/src/main/java/com/polarstation/diary10/fragment/MainFragmentCallBack.java
package com.polarstation.diary10.fragment;

import android.app.Activity;

import com.polarstation.diary10.data.DiaryRecyclerViewAdapter;

public interface MainFragmentCallBack {
    void replaceFragment(int type);
    void quitApp();
    Activity getActivity();
    void setNavigationViewDisabled();
    void notifyAdapter(DiaryRecyclerViewAdapter adapter);
}

File: Diary10/app/src/main/java/com/polarstation/diary10/fragment/MainFragmentCallback.kt
package com.polarstation.diary10.fragment

import android.app.Activity
import com.polarstation.diary10.data.DiaryRecyclerViewAdapter

interface MainFragmentCallback{
    fun replaceFragment(type: Int)
    fun quitApp()
    fun getActivity(): Activity
    fun setNavigationViewDisabled()
    fun notifyAdapter(adapter: DiaryRecyclerViewAdapter)
}